mlfinlab/portfolio_optimization/hcaa.py

```diff
@@ -224,13 +232,130 @@ def _quasi_diagnalization(self, num_assets, curr_index):
 
         return (self._quasi_diagnalization(num_assets, left) + self._quasi_diagnalization(num_assets, right))
 
+    def _recursive_bisection(self,
+                             expected_asset_returns,
+                             asset_returns,
+                             covariance_matrix,
+                             assets,
+                             allocation_metric,
+                             optimal_num_clusters,
+                             confidence_level):
+        # pylint: disable=bad-continuation, too-many-locals
+        """
+        Recursively assign weights to the clusters - ultimately assigning weights to the individual assets.
+
+        :param expected_asset_returns: (list) a list of mean asset returns (mu)
```

Comment:
Please fix all the grammar that was changed. @PanPip just spent a week updating the docs for the convention to be reasonably uniform.

Reply:
Done